### ISSUE TITLE
Missing SC2086 on commands beginning with input direction

### ISSUE THREAD

**Issue by FrankHB** [koalaman/shellcheck#2811]
#### For bugs
- Rule Id (if any, e.g. SC2046): SC2086
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/env bash
app="/some path/to-app"
< filename $app # missing SC2086
mapfile < filename $app # OK
mapfile < <(< filename $app) # missing SC2086
```

#### Here's what shellcheck currently says:

$ shellcheck myscript
 
[Line 4:](javascript:setPosition(4, 20))
mapfile < filename $app # OK
                   ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.

Did you mean: ([apply this](javascript:applyFixIndex([0])), apply [all SC2086](javascript:applyFixCode(2086)))
mapfile < filename "$app" # OK

#### Here's what I wanted or expected to see:

There should be 3 instances of SC2086 Lines 3, 4 and 5 should all 

#### Notes

The real case is from:

https://github.com/FrankHB/YSLib/blob/596d7a8e93ca2418b0b9fb0ef57d74f03d84867a/Tools/Scripts/PatchRevision.sh#L23

**Comment by wileyhy** [comment]
The OP's example is a case where an absolute path contains whitespace.
Executing `$cmd` without double quotes would cause an error.

$ touch file
$ cmd='./bin - test/rm'
$ set -x ; $cmd file ; echo $? ; set -
+ ./bin - test/rm file
bash: ./bin: No such file or directory
+ echo 127
127

This is a reasonable example of how relying on word splitting actually is
very fragile.

However, bash can locate and execute commands in directories in PATH the
names of which contain whitespace, evidently (on Android, anyway).

$ pwd
/data/data/com.termux/files/home
$ mkdir './bin - test'
$ cp "$( command -v rm )" ./bin\ -\ test/
$ PATH="./bin - test:$PATH"
$ declare -p PATH                      declare -x PATH="./bin -
test:/data/data/com.termux/files/usr/bin:/data/data/com.termux/files/usr/bin/applets"
$ hash -r
$ hash
hash: hash table empty
$ hash -p "$( type -P rm )" rm
$ hash
hits    command
   0    ./bin - test/rm
$ touch foo
$ set -x ; rm foo ; echo $? ; set -
+ rm foo
+ echo 0
0
+ set -
$ hash
hits    command
   1    /data/data/com.termux/files/usr/bin/touch
   1    ./bin - test/rm

Cheers,
Wiley

**Comment by koalaman** [comment]
You're right about the problem, of course.

ShellCheck intentionally excludes SC2086 from the first word in a command (with or without redirection). This is due to the empirical observation that such a warning is significantly more likely to break a script expanding a command+args, than it is to fix a script running an executable with spaces in the path.

I would prefer emitting warnings the way you describe it, but a concession was made to improve accuracy and reduce noise.

**Comment by FrankHB** [comment]
> You're right about the problem, of course.
> 
> ShellCheck intentionally excludes SC2086 from the first word in a command (with or without redirection). This is due to the empirical observation that such a warning is significantly more likely to break a script expanding a command+args, than it is to fix a script running an executable with spaces in the path.
> 
> I would prefer emitting warnings the way you describe it, but a concession was made to improve accuracy and reduce noise.

So perhaps better explicitly mention the rationale of the intention in the wiki page?
